### DISCUSSION TITLE
New flag to config API keys from the command line

### DISCUSSION THREAD

**Discussion by vzamanillo** [projectdiscovery/subfinder#663] [Ideas]
Hey,

Many new people to Subfinder are not clear on how to configure the API keys to get some sources working because they are not familiarized with YAML format, maybe adding a new feature to do that from the command line could be useful.

The concept:

![imagen](https://user-images.githubusercontent.com/10209695/193639025-b9e1fddf-fdf6-444f-ac29-5065ee00571e.png)

The command:

`subfinder -sa virustotal -kL key1,key2,keyn`

![imagen](https://user-images.githubusercontent.com/10209695/193639258-50f5c488-7abb-4ef0-b882-88d0aa33110b.png)

Thoughts?

**Comment by forgedhallpass** [comment]
If people have problems with setting the keys, then we rather need to update the documentation. It's not that complicated...

I don't think accepting keys from the CLI makes sense, for multiple reasons:
* it's inconvenient to copy paste the keys over and over again
* usually it's not recommended to pass in secrets in plain text because they get stored in the shell history
* extra effort and maintenance

Besides the above mentioned it would require refactoring other parts of the application as well. 

E.g. a user might have multiple keys for a provider, and multiple providers, hence he'd need to pass in `-kL source1:secret1 -kL source1:secret2 -kL source2:secret3` and so on. It would very fast become unreadable and painful to edit.

**Reply by vzamanillo** [reply]
My idea was about a one time config api key configuration, not pasting the keys over and over again, I mean, a way to store the keys without editing the config file, but you are completely right, it is risky at all and hard to maintain.
